epictrack-web/src/components/workPlan/status/StatusView/index.tsx

```diff
@@ -26,7 +27,11 @@ const StatusView = () => {
           onAddNewClickHandler={() => onAddButtonClickHandler()}
         />
       </When>
-      <When condition={isStatusOutOfDate()}>
+      <When
+        condition={isStatusOutOfDate(
+          statuses.find((status) => status.is_approved)
```

Comment:
can we have a variable defined for all of this const statusOutOfDate = isStatusOutOfDate(......) and have it defined outside the return. in the When it would be just condition={statusOutOfDate}. Same idea for the rest of similar instances